src/components/Portal/PortalInner.tsx

```diff
@@ -0,0 +1,48 @@
+import * as PropTypes from 'prop-types'
+import { invoke } from 'lodash'
+import { Component } from 'react'
+import { createPortal } from 'react-dom'
+import { Extendable } from '../../../types/utils'
+
+export interface IPortalProps {
+  onMount?: (props: IPortalProps) => void
+  onUnmount?: (props: IPortalProps) => void
+}
+
+/**
+ * An inner component that allows you to render children outside their parent.
+ */
+class PortalInner extends Component<Extendable<IPortalProps>, any> {
+  public static propTypes = {
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /**
+     * Called when the portal is mounted on the DOM
+     *
+     * @param {object} data - All props.
+     */
+    onMount: PropTypes.func,
+
+    /**
+     * Called when the portal is unmounted from the DOM
+     *
+     * @param {object} data - All props.
+     */
+    onUnmount: PropTypes.func,
+  }
+
+  public componentDidMount() {
+    invoke(this.props, 'onMount', { ...this.props })
```

Comment:
Since we're not adding / changing any props here, not sure that a shallow clone is required.  We can save on the cycles by just passing `this.props`.

Reply:
made the change as suggested